src/people.rs

```diff
@@ -257,26 +421,56 @@ impl ContextPeopleExt for Context {
         property: T,
         value: T::Value,
     ) {
+        assert!(!T::is_derived(), "Cannot set a derived property");
         let data_container = self.get_data_container(PeoplePlugin)
             .expect("PeoplePlugin is not initialized; make sure you add a person before accessing properties");
 
-        let current_value = *data_container.get_person_property_ref(person_id, property);
-        let previous_value = match current_value {
-            Some(current_value) => current_value,
+        let current_cached_value = *data_container.get_person_property_ref(person_id, property);
+        let previous_value = match current_cached_value {
+            Some(value) => value,
             None => {
-                let initialize_value = T::initialize(self, person_id);
+                let initialize_value = T::compute(self, person_id);
                 data_container.set_person_property(person_id, property, initialize_value);
                 initialize_value
             }
         };
 
+        // Temporarily remove dependency properties since we need mutable references
+        // to self during callback execution
+        let deps_temp = {
+            let data_container = self.get_data_container(PeoplePlugin).unwrap();
+            let mut dependencies = data_container.dependency_map.borrow_mut();
+            dependencies.get_mut(&TypeId::of::<T>()).map(std::mem::take)
+        };
+
+        let mut dependency_event_callbacks = Vec::new();
+        if let Some(mut deps) = deps_temp {
+            // If there are dependencies, set up a bunch of callbacks with the
+            // current value
+            for dep in &mut deps {
+                dep.add_event_callback(self, person_id, &mut dependency_event_callbacks);
```

Comment:
I would call this `dependency_changed()` so that (for instance) you can handle indexing in the same cb if you want to.